icepyx/tests/test_Earthdata.py

```diff
@@ -0,0 +1,123 @@
+#!/usr/bin/env python
+u"""
+test_Earthdata.py (05/2021)
+"""
+import os
+import re
+import netrc
+import pytest
+import shutil
+import getpass
+import warnings
+import posixpath
+# from icepyx.core.Earthdata import Earthdata
```

Comment:
So the tests in test_Earthdata.py are now all testing the mock Earthdata class declared at L49. Is the intention to only test things using the mock on Travis CI, and that we need to uncomment this line manually if we want to test the actual CMR API?

Reply:
that is how it is currently written.  I thought about making it a command line option (like `--use-api`, `--full-integration` or something) and changing the mock definition to `_Earthdata` or something similar.

Reply:
> that is how it is currently written. I thought about making it a command line option (like `--use-api`, `--full-integration` or something) and changing the mock definition to `_Earthdata` or something similar.

This seems rather complicated (and possibly hard to maintain). I don't have a good solution either, but wondering if there's a built-in pytest plugin that can be toggled on/off to do this sort of thing. Or what about scheduling the test_Earthdata.py tests to run only once a day or once per week if pinging the NSIDC servers constantly is an issue?

Side note: Using 'mocks' often seems to be a last resort (in my limited experience of using them in unit tests), so I just want to make they're used here because other options have been exhausted already.

Reply:
That would be in travis to only run the full integration tests occasionally.  I think that is what @JessicaS11 is doing already for some of the testing suite.  Github actions has cron scheduling as well.

Adding the mock was suggested just to keep the testing suite isolated.  I doubt pinging the CMR system for a token on PRs would be a problem (except for cases when the system is down for scheduled maintenance).  Would come up with a workaround in those cases. 🤷 

Might be worth some further discussion.

Reply:
@tsutterley captured my thoughts correctly. I think we should keep `test_Earthdata.py` on a mock to keep the testing isolated. Then we can tear into `behind_NSIDC_API_login.py` (including a rename that includes "test" - there was a major code refactor early on and not adding test to the filename is my cue that these tests/files weren't updated to the new code structure) to do the actual pinging of NSIDC (which is already set up to run as a cron job via Travis, but was commented out to keep it from running until the tests were updated).

I will start into that updating, since I think that's one of the other pieces keeping this PR from being merged.

Reply:
see #209